### ISSUE TITLE
508-defect-1❗ Launchblocker [SCREENREADER, AXE-CORE, SEMANTIC MARKUP]: `<ul>` and `<ol>` MUST only directly contain `<li>`, `<script>` or `<template>` elements

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#8961]
# [508-defect-1 :exclamation: Launchblocker](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-1)

**Feedback framework**

- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

`<ul>` and `<ol>` **must** only directly contain `<li>`, `<script>` or `<template>` elements. Lists **must** be marked up correctly, meaning they must not contain content elements other than li elements.

### Why it Matters

> **Screen readers have a specific way of announcing lists. This feature makes lists clearer to understand, but will only work if lists are properly structured.**
> 
> When content elements other than list items are contained within a set of list elements, screen readers cannot inform the listener that they are listening to items within the list.
> 
> For a list to be valid, it must have both parent elements (a set of ul elements or a set of ol elements) and child elements (declared inside of these tags using the li element), and any other content elements are invalid.
> 
> Although some non-content elements such as script, template, style, meta, link, map, area, and datalist are permitted within lists, content elements other than li are not permitted.

## Point of Contact

**VFS Point of Contact:** Jennifer

## Acceptance Criteria

As a screen reader user, I want to read and understand the content of a list.

## Environment

* Operating System: all
* Browser: all
* Screenreading device: any
* Server destination: https://staging.va.gov/pittsburgh-health-care/health-services/

## Steps to Recreate

1. Enter https://staging.va.gov/pittsburgh-health-care/health-services/ in browser
2. Resize browser to less than 768px wide
3. Run the axe browser scan on the page
4. Verify the issue appears: "`<ul>` and `<ol>` must only directly contain `<li>`, `<script>` or `<template>` elements"

## Possible Fixes (optional)

**Consider** moving the inline CSS into styles that apply to the elements, rather than repeating styling within each element.

### Current code
```diff
<ul id="va-sidenav-ul-container" class="va-sidenav vads-u-margin-top--0 vads-u-padding--0">
!  <div class="line medium-screen:vads-u-display--none va-sidenav-display-onclick-line">
  </div>
  <li class="va-sidenav-level-1">
    <h2 class="va-sidenav-item-label, vads-u-font-family--sans va-sidenav-item-label-bold" style="padding-left: 20px; font-size: 15px; text-transform: uppercase;">
      Services and locations
    </h2>
    <ul>
      <li class="va-sidenav-level-2 active">
        <a aria-label="Health services" class="va-sidenav-item-label open" href="/pittsburgh-health-care/health-services" rel="noopener noreferrer" style="padding-left: 20px;"><span class=""> Health services </span></a>
      </li>
      <li class="va-sidenav-level-2">
        <a aria-label="Locations" class="va-sidenav-item-label" href="/pittsburgh-health-care/locations" rel="noopener noreferrer" style="padding-left: 20px;"><span class=""> Locations </span></a>
      </li>
    </ul>
  </li>
  <li class="va-sidenav-level-1">
    <h2 class="va-sidenav-item-label, vads-u-font-family--sans va-sidenav-item-label-bold" style="padding-left: 20px; font-size: 15px; text-transform: uppercase;">
      News and events
    </h2>
    <ul>
      <li class="va-sidenav-level-2">
        <a aria-label="Events" class="va-sidenav-item-label" href="/pittsburgh-health-care/events" rel="noopener noreferrer" style="padding-left: 20px;"><span class=""> Events </span></a>
      </li>
      <li class="va-sidenav-level-2">
        <a aria-label="News releases" class="va-sidenav-item-label" href="/pittsburgh-health-care/news-releases" rel="noopener noreferrer" style="padding-left: 20px;"><span class=""> News releases </span></a>
      </li>
      <li class="va-sidenav-level-2">
        <a aria-label="Stories" class="va-sidenav-item-label" href="/pittsburgh-health-care/stories" rel="noopener noreferrer" style="padding-left: 20px;"><span class=""> Stories </span></a>
      </li>
    </ul>
  </li>
  <li class="va-sidenav-level-1">
    <h2 class="va-sidenav-item-label, vads-u-font-family--sans va-sidenav-item-label-bold" style="padding-left: 20px; font-size: 15px; text-transform: uppercase;">About VA Pittsburgh</h2>
    <ul>
      <li class="va-sidenav-level-2">
        <a aria-label="About us" class="va-sidenav-item-label" href="/pittsburgh-health-care/about" rel="noopener noreferrer" style="padding-left: 20px;"><span class=""> About us </span></a>
      </li>
      <li class="va-sidenav-level-2">
        <a aria-label="Work with us" class="va-sidenav-item-label" href="/pittsburgh-health-care/work-with-us" rel="noopener noreferrer" style="padding-left: 20px;"><span class=""> Work with us </span></a>
      </li>
      <li class="va-sidenav-level-2">
        <a aria-label="Programs" class="va-sidenav-item-label" href="/pittsburgh-health-care/programs" rel="noopener noreferrer" style="padding-left: 20px;"><span class=""> Programs </span></a>
      </li>
      <li class="va-sidenav-level-2">
        <a aria-label="Research" class="va-sidenav-item-label" href="/pittsburgh-health-care/research" rel="noopener noreferrer" style="padding-left: 20px;"><span class=""> Research </span></a>
      </li>
      <li class="va-sidenav-level-2">
        <a aria-label="Policies" class="va-sidenav-item-label" href="/pittsburgh-health-care/policies" rel="noopener noreferrer" style="padding-left: 20px;"><span class=""> Policies </span></a>
      </li>
      <li class="va-sidenav-level-2">
        <a aria-label="Contact us" class="va-sidenav-item-label" href="/pittsburgh-health-care/contact-us" rel="noopener noreferrer" style="padding-left: 20px;"><span class=""> Contact us </span></a>
      </li>
    </ul>
  </li>
</ul>
```

### Possible fix
```diff

<!-- styles are loosely documented and likely not accurate -->
<style>

#va-sidenav-ul-container h2 { 
  border-top: 1px solid #d6d7d9; 
  font-size: 15px; 
  margin: 10px 1.5rem;
  padding-left: 20px; 
  padding-top: 1.5rem; 
  text-transform: uppercase; }
.va-sidenav-level-2 a {
  padding-left: 20px; }

</style>


<ul id="va-sidenav-ul-container" class="va-sidenav vads-u-margin-top--0 vads-u-padding--0">
+ <!-- the div.line must be removed; add the line in CSS as a border-top for the h2 instead -->
  <li class="va-sidenav-level-1">
    <h2 class="va-sidenav-item-label, vads-u-font-family--sans va-sidenav-item-label-bold">
      Services and locations
    </h2>
    <ul>
      <li class="va-sidenav-level-2 active">
        <a aria-label="Health services" class="va-sidenav-item-label open" href="/pittsburgh-health-care/health-services" rel="noopener noreferrer"><span class=""> Health services </span></a>
      </li>
      <li class="va-sidenav-level-2">
        <a aria-label="Locations" class="va-sidenav-item-label" href="/pittsburgh-health-care/locations" rel="noopener noreferrer"><span class=""> Locations </span></a>
      </li>
    </ul>
  </li>
  <li class="va-sidenav-level-1">
    <h2 class="va-sidenav-item-label, vads-u-font-family--sans va-sidenav-item-label-bold">
      News and events
    </h2>
    <ul>
      <li class="va-sidenav-level-2">
        <a aria-label="Events" class="va-sidenav-item-label" href="/pittsburgh-health-care/events" rel="noopener noreferrer"><span class=""> Events </span></a>
      </li>
      <li class="va-sidenav-level-2">
        <a aria-label="News releases" class="va-sidenav-item-label" href="/pittsburgh-health-care/news-releases" rel="noopener noreferrer"><span class=""> News releases </span></a>
      </li>
      <li class="va-sidenav-level-2">
        <a aria-label="Stories" class="va-sidenav-item-label" href="/pittsburgh-health-care/stories" rel="noopener noreferrer"><span class=""> Stories </span></a>
      </li>
    </ul>
  </li>
  <li class="va-sidenav-level-1">
    <h2 class="va-sidenav-item-label, vads-u-font-family--sans va-sidenav-item-label-bold">About VA Pittsburgh</h2>
    <ul>
      <li class="va-sidenav-level-2">
        <a aria-label="About us" class="va-sidenav-item-label" href="/pittsburgh-health-care/about" rel="noopener noreferrer"><span class=""> About us </span></a>
      </li>
      <li class="va-sidenav-level-2">
        <a aria-label="Work with us" class="va-sidenav-item-label" href="/pittsburgh-health-care/work-with-us" rel="noopener noreferrer"><span class=""> Work with us </span></a>
      </li>
      <li class="va-sidenav-level-2">
        <a aria-label="Programs" class="va-sidenav-item-label" href="/pittsburgh-health-care/programs" rel="noopener noreferrer"><span class=""> Programs </span></a>
      </li>
      <li class="va-sidenav-level-2">
        <a aria-label="Research" class="va-sidenav-item-label" href="/pittsburgh-health-care/research" rel="noopener noreferrer"><span class=""> Research </span></a>
      </li>
      <li class="va-sidenav-level-2">
        <a aria-label="Policies" class="va-sidenav-item-label" href="/pittsburgh-health-care/policies" rel="noopener noreferrer"><span class=""> Policies </span></a>
      </li>
      <li class="va-sidenav-level-2">
        <a aria-label="Contact us" class="va-sidenav-item-label" href="/pittsburgh-health-care/contact-us" rel="noopener noreferrer"><span class=""> Contact us </span></a>
      </li>
    </ul>
  </li>
</ul>
```

## WCAG or Vendor Guidance (optional)

* [Deque University documentation](https://dequeuniversity.com/rules/axe/3.5/list?application=AxeChrome)

* [Using ol, ul, and dl for lists or groups of links](http://www.w3.org/TR/WCAG20-TECHS/H48.html)

**Comment by jenstrickland** [comment]
I just took a look and this appears to still be there.

**Comment by mmiddaugh** [comment]
should be ready for your review @jenstrickland

**Comment by jenstrickland** [comment]
Validated.
